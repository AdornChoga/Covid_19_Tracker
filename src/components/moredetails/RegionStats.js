import React from 'react';
import PropTypes from 'prop-types';
import classes from './MoreDetails.module.css';
import Header from '../header/Header';
import CountryHeader from './CountryHeader';
import Details from './Details';
import createJSProperties from '../../logic/CreateJSProperties';

const RegionStats = (props) => {
  const { region, country } = props;
  const regionStatistics = createJSProperties(region);
  const { name: regionName } = regionStatistics;

  return (
    <div className={classes.main_container}>
      <Header />
      <CountryHeader country={country} />
      <h4>
        Region:&nbsp;&nbsp;&nbsp;
        {regionName}
      </h4>
      <Details details={regionStatistics} />
    </div>
  );
};

RegionStats.propTypes = {
  country: PropTypes.shape({}).isRequired,
  region: PropTypes.shape({
    name: PropTypes.string.isRequired,
  }).isRequired,
};

export default RegionStats;
