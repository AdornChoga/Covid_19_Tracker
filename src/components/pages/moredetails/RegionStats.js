import React from 'react';
import PropTypes from 'prop-types';
import Header from '../../header/Header';
import Country from '../../countries/Country';

const RegionStats = (props) => {
  const { region, country } = props;
  const {
    name: regionName, today_new_open_cases: todayNewCases, today_new_recovered: todayRecoveries,
    today_new_deaths: todayDeaths, today_confirmed: totalCasesRecorded,
    today_recovered: totalRecoveriesRecorded, today_deaths: totalDeathsRecorded,
  } = region;

  return (
    <>
      <Header />
      <div>
        <Country country={country} />
        <h3>
          Region:&nbsp;&nbsp;&nbsp;
          {regionName}
        </h3>
        <ul>
          <li>
            Today&apos;s new cases:&nbsp;&nbsp;&nbsp;
            {todayNewCases}
          </li>
          <li>
            Today&apos;s new recoveries:&nbsp;&nbsp;&nbsp;
            {todayRecoveries}
          </li>
          <li>
            Today&apos;s new deaths:&nbsp;&nbsp;&nbsp;
            {todayDeaths}
          </li>
          <li>
            Total cases recorded:&nbsp;&nbsp;&nbsp;
            {totalCasesRecorded}
          </li>
          <li>
            Total recoveries recorded:&nbsp;&nbsp;&nbsp;
            {totalRecoveriesRecorded}
          </li>
          <li>
            Total deaths recorded:&nbsp;&nbsp;&nbsp;
            {totalDeathsRecorded}
          </li>
        </ul>
      </div>
    </>
  );
};

RegionStats.propTypes = {
  country: PropTypes.shape({}).isRequired,
  region: PropTypes.shape({
    name: PropTypes.string.isRequired,
    today_new_open_cases: PropTypes.number,
    today_new_recovered: PropTypes.number,
    today_new_deaths: PropTypes.number,
    today_confirmed: PropTypes.number,
    today_recovered: PropTypes.number,
    today_deaths: PropTypes.number,
  }).isRequired,
};

export default RegionStats;
