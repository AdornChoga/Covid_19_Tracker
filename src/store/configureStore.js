import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import statistics from './statistics/statistics';

const reducer = combineReducers({
  statistics,
});

const store = createStore(reducer, applyMiddleware(logger, thunk));

export default store;
