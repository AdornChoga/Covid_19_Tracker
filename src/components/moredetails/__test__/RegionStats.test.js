import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../../store/configureStore';
import RegionStats from '../RegionStats';

it('should create region stats component snapshot', () => {
  const country = {
    name: 'Canada',
    image: 'image.png',
    todayOpenCases: 4949,
    regions: [],
  };

  const region = {
    name: '',
    id: '',
    regions: [],
    today_new_open_cases: 8599,
    today_new_recovered: 858,
    today_new_deaths: 858,
    today_open_cases: 303,
    today_confirmed: 304,
    today_recovered: 134,
    today_deaths: 455,
  };
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <RegionStats region={region} country={country} />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
