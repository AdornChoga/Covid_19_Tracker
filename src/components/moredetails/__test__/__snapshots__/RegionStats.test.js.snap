// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create region stats component snapshot 1`] = `
<div
  className="main_container"
>
  <header>
    <svg
      className="prev_page"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
      />
    </svg>
    <h1>
      Today's
         Statistics
    </h1>
    <div>
      <form>
        
        <button
          onClick={[Function]}
          type="submit"
        >
          <svg
            className="search_icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
            />
          </svg>
        </button>
        <br />
        <ul
          className="search_results"
        />
      </form>
      <svg
        className="filter"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M487.976 0H24.028C2.71 0-8.047 25.866 7.058 40.971L192 225.941V432c0 7.831 3.821 15.17 10.237 19.662l80 55.98C298.02 518.69 320 507.493 320 487.98V225.941l184.947-184.97C520.021 25.896 509.338 0 487.976 0z"
        />
      </svg>
    </div>
  </header>
  <div
    className="country"
  >
    <img
      alt="map"
      src="image.png"
    />
    <div>
      <span
        className="country_name"
      >
        Canada
      </span>
      <span
        className="country_open_cases"
      >
        Open Cases:   
        4949
      </span>
    </div>
  </div>
  <h4>
    Region:   
    
  </h4>
  <ul
    className="statistics"
  >
    <li>
      Today's new cases:   
      8599
    </li>
    <li>
      Today's new recoveries:   
      858
    </li>
    <li>
      Today's new deaths:   
      858
    </li>
    <li>
      Total cases recorded:   
      304
    </li>
    <li>
      Total recoveries recorded:   
      134
    </li>
    <li>
      Total deaths recorded:   
      455
    </li>
  </ul>
</div>
`;
